### ISSUE TITLE
Binary `contents_pillar` with file.managed raises UnicodeDecodeError

### ISSUE THREAD

**Issue by epelc** [saltstack/salt#46909]
### Description of Issue/Question
Trying to use `file.managed` with a binary content from `contents_pillar` in 2018.3.0 doesn't work. This use to work in 2017.7.x

mystate.sls
```
somefile:
  file.managed:
    - contents_pillar: files:app:mykey
    - contents_newline: False
```

```
salt '*' state.sls mystate
```

Error
```
       ID: /srv/server/session-ticket-keys/0.key
    Function: file.managed
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "/usr/lib/python2.7/dist-packages/salt/state.py", line 1878, in call
                  **cdata['kwargs'])
                File "/usr/lib/python2.7/dist-packages/salt/loader.py", line 1823, in wrapper
                  return f(*args, **kwargs)
                File "/usr/lib/python2.7/dist-packages/salt/states/file.py", line 2267, in managed
                  isinstance(use_contents, six.string_types) and '\0' in use_contents
              UnicodeDecodeError: 'ascii' codec can't decode byte 0xf5 in position 1: ordinal not in range(128)
     Started: 17:29:34.221921
    Duration: 4.871 ms
     Changes:   

```

### Setup
(Please provide relevant configs and/or SLS files (Be sure to remove sensitive info).)

### Steps to Reproduce Issue
(Include debug logs if possible and relevant.)

### Versions Report
(Provided by running `salt --versions-report`. Please also mention any differences in master/minion versions.)
```
Salt Version:
           Salt: 2018.3.0
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.6.0
      docker-py: 3.2.1
          gitdb: 0.5.4
      gitpython: 0.3.2 RC1
          ioflo: Not Installed
         Jinja2: 2.10
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 0.9.1
   msgpack-pure: Not Installed
 msgpack-python: 0.5.6
   mysql-python: 1.2.3
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.6 (default, Nov 23 2017, 15:49:48)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 17.0.0
           RAET: Not Installed
          smmap: 0.8.2
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.1.6
 
System Versions:
           dist: Ubuntu 14.04 trusty
         locale: UTF-8
        machine: x86_64
        release: 4.4.0-96-generic
         system: Linux
        version: Ubuntu 14.04 trusty
```

**Comment by epelc** [comment]
I found a workaround.

You can use file.managed to generate a base64 file of a different name then use `cmd.run` to basically decode your file that salt can manage back into your binary file.

**Comment by terminalmage** [comment]
Should be fixed via https://github.com/saltstack/salt/pull/47041, I plan on adding a regression test for this tomorrow.

**Comment by terminalmage** [comment]
Test added to the PR.

**Comment by Cecolo** [comment]
@epelc +1: Can you post the workaround? Thanks

**Comment by rallytime** [comment]
Closed via #47041
